### PR TITLE
Properly annotate function pointers so stack usage is properly calculated for multiple instances

### PR DESCRIPTION
**Pull request by ed-xmos**
Looks like fptr groups were not fully implemented, causing stack corruption in my ASRC implementation which uses multiple threads. 
Also, the head of develop didn't pass lib tests so fixed that too. I have temporarily given this version 2.5.0 since there is a fix (patch) plus a new feature (minor)

### CODE REVIEW

**Comment by xross** [comment]
We need to fix these lib tests (or ignore the fails) - we don't want to have version numbers in the git history that don't exist.

**Comment by ed-xmos** [comment]
> We need to fix these lib tests (or ignore the fails) - we don't want to have version numbers in the git history that don't exist.

OK - this is the way the software team have always done it so this is a different practice to what I am used to. Does it really matter as long as the changelog is correct (semantic version-wise) and no tagged release has been made? We need a green Jenkins run before we can merge and disabling tests doesn't feel right.

**Comment by xross** [comment]
Understand. Ive made (another) note to fix the process/scripts

**Review by xross — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+2.5.0
+-----
+
+  * ADDED: Support for XCommon CMake build system
+  * FIXED: Function pointer annotation avoid stack corruption when using
+    multiple instances of SSRC or ASRC.
+
+  * Changes to dependencies:
+
+    - lib_logging: Removed dependency
```
[the remark this answers is not in the document]
Comment (danielpieczko) [reply CHANGELOG.rst:13]: Saying that, it doesn't look like there are any assert calls in this library, so lib_xassert could be remove from the dependency lists in module_build_info and lib_build_info.cmake.

[the remark this answers is not in the document]
Comment (ed-xmos) [reply CHANGELOG.rst:13]: Yes, the changelog script insisted that was the case

**Pull request merged**